vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.ts

```diff
@@ -215,24 +215,40 @@ import { GridFlowSelectionColumn } from "./vaadin-grid-flow-selection-column.js"
           return false;
         });
 
+        grid.$connector.debounceRootRequest = tryCatchWrapper(function (page) {
+          const delay = grid._hasData ? rootRequestDelay : 0;
+
+          rootRequestDebouncer = Debouncer.debounce(rootRequestDebouncer, timeOut.after(delay), () => {
+            grid.$connector.fetchPage(
+              (firstIndex, size) => grid.$server.setRequestedRange(firstIndex, size),
+              page,
+              root
+            );
+          });
+        });
+
         grid.$connector.flushParentRequests = tryCatchWrapper(function () {
-          let pendingFetches = parentRequestQueue.splice(0, parentRequestBatchMaxSize);
+          const pendingFetches = [];
+
+          parentRequestQueue.splice(0, parentRequestBatchMaxSize).forEach(({ parentKey, page }) => {
+            grid.$connector.fetchPage(
+              (firstIndex, size) => pendingFetches.push({ parentKey, firstIndex, size }),
+              page,
+              parentKey
+            );
+          });
 
           if (pendingFetches.length) {
             grid.$server.setParentRequestedRanges(pendingFetches);
-            return true;
           }
-          return false;
         });
 
-        grid.$connector.beforeParentRequest = tryCatchWrapper(function (firstIndex, size, parentKey) {
-          // add request in queue
-          parentRequestQueue.push({
-            firstIndex: firstIndex,
-            size: size,
-            parentKey: parentKey
-          });
-
+        grid.$connector.debounceParentRequest = tryCatchWrapper(function (parentKey, page) {
+          // Remove any pending requests for the same parentKey.
+          parentRequestQueue = parentRequestQueue.filter((request) => request.parentKey !== parentKey);
```

Comment:
**note:** It now seems that a more appropriate data structure to store scheduled requests would be Map rather than Array. However, I didn't change it to minimize the impact of the changes, considering that the PR will likely need to be backported to early versions.